compiler/fir/fir2ir/src/org/jetbrains/kotlin/fir/backend/Fir2IrLocalStorage.kt

```diff
@@ -40,7 +40,7 @@ class Fir2IrLocalStorage {
     }
 
     fun getLocalClass(classId: ClassId): IrClass? {
-        return localClassCache.entries.find { (firClass, _) -> firClass.classId == classId }?.value
+        return localClassCache.entries.find { (firClass, _) -> firClass.classId === classId }?.value
```

Comment:
Possible alternative solution is to remove this function at all and use `getLocalClass(firClass)` instead. This probably allows to use single `ANONYMOUS_CLASS_ID` as before.

Let's leave is as is at the moment, I will discuss this place with @semoro shortly and may be will perform this small refactoring myself.

Reply:
Sounds good to me. I'll look at another module meanwhile.